src/cs_main.h

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_CS_MAIN_H
+#define BITCOIN_CS_MAIN_H
+
+#include <sync.h>
+
+/**
+ * Mutex to guard access to validation specific variables, such as reading
+ * or changing the chainstate.
+ *
+ * This may also need to be locked when updating the transaction pool, e.g. on
+ * AcceptToMemoryPool. See CTxMemPool::cs comment for details.
+ *
+ * The transaction pool has a separate lock to allow reading from it and the
+ * chainstate at the same time.
+ */
+static RecursiveMutex cs_main;
```

Comment:
In commit "refactor: add cs_main.h" (7e2a68204ecbc15a6f661075a34300fe5c101232)

I think this still needs to be `extern` not `static`. Otherwise there would be a different instance of `cs_main` variable in each translation unit and locking one of the `cs_main` variables would not lock all the other ones.

Reply:
Making this `extern` causes link errors on multiple platforms, will take another look.

Reply:
> Making this `extern` causes link errors on multiple platforms, will take another look.

Right you would also need to add `RecursiveMutex cs_main;` without `extern` to a `.cpp` file like `cs_main.cpp`, or undelete it from `validation.cpp` to actually instantiate the variable and create the linker symbol. Declaring a variable without `static` or `extern` will creates a linker symbol. Declaring a variable `extern` just tells the compiler the type of the variable and makes it look for an exisiting symbol to import. Declaring a variable `static` makes the compiler create a private variable in the current translation unit without importing or exporting any symbol.

Reply:
Yep, you are completely correct. My oversight here.